logging/slog_logger.go

```diff
@@ -1,75 +1,128 @@
 package logging
 
 import (
+	"context"
 	"fmt"
 	"log/slog"
 	"os"
+	"runtime"
+	"time"
+)
+
+type SlogHandlerType string
+
+const (
+	SLogHandlerTypeJson     SlogHandlerType = "json" // prints debug and above
+	SLogHandlerTypeJsonText SlogHandlerType = "text" // prints info and above
 )
 
 type SLogger struct {
-	logger *slog.Logger
+	*slog.Logger
 }
 
 var _ Logger = (*SLogger)(nil)
 
-func NewSlogLogger(env LogLevel) Logger {
+func NewSlogTextLogger(env LogLevel) *SLogger {
+	var handler slog.Handler
+	if env == Production {
```

Comment:
Not sure if it makes sense to filter out logs based on production vs development. Sometimes we need to output debug logs in our production environment so we can properly debug production issues.

Reply:
Wouldn't it make sense to just make env the levels themselves from slog?

Reply:
Yeah we should.. we were using the log level names from zapper, no idea why they only have prod and dev. I would vote to just switch everything to slog, but this might break some client code. You fine with this @shrimalmadhur ?
Honestly eventually we should probably just migrate our zap logger to using the slog handler https://github.com/uber-go/zap/tree/master/exp/zapslog so that everything's consistent. Thoughts @shrimalmadhur ?

Reply:
@dmanc are you currently switching the log level dynamically? That's an option we could add, by exposing some REST endpoint or whatever.

Reply:
@dmanc changed the constructors in [bded386](https://github.com/Layr-Labs/eigensdk-go/pull/127/commits/bded3869144bcfd7124ac19ebd63ac753d9f0b19)
Let me know what you think

Reply:
We're not switching dynamically during runtime but that sounds like a good idea! Right now it's just config and we need to restart.

Reply:
How helpful would this dynamic switching be? Like how important is it? Aka can we merge this without?

Reply:
Not important right now

Reply:
Ok will resolve this thread then. Let's revisit this dynamic switching later.